.github/workflows/ci.yml

```diff
@@ -0,0 +1,33 @@
+name: Node.js CI
+on: [pull_request]
+
+jobs:
+  git_checks:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          architecture: "x64"
+      - name: Use pnpm actions
+        uses: pnpm/action-setup@v2
+        with:
+          version: 8
+      - name: Initialize project
+        run: pnpm install --frozen-lockfile
+      - name: Ensure linear-history
+        run: git merge-base --is-ancestor "origin/${GITHUB_BASE_REF}" HEAD
+      - name: Check for fixup or WIP commits
+        run: |
+          ! git log "origin/${GITHUB_BASE_REF}..HEAD" | egrep ' fixup! |wip|Wip|WIP'
+      - name: Run lint
+        run: pnpm lint
+      - name: Run check-format
+        run: pnpm check-format
+      - name: Run type-check
```

Comment:
"Run type checks" as the name of the package.json scripts is an internal matter and is neither necessarily known nor of interest to the person reading the output of the github actions.